project/backend/src/test/java/com/google/sps/JsonToRestaurantParserTest.java

```diff
@@ -0,0 +1,101 @@
+  private static final String RESTAURANT_NAME = "RESTAURANT_A";
+  private static final String PLACE_ID = "A12345";
+  private static final String VICINITY = "10 Main Street";
+  private static final Map<String, Double> COORDS = new HashMap<String, Double>() {{
+    put("lat", 10.0);
+    put("lng", 20.0);
+  }};
+
+  @Test
+  public void allRequiredAndOptionalFields() throws JSONException {
+    JSONObject geometry = new JSONObject().put("location", COORDS);
+    JSONObject body = new JSONObject()
+      .put("place_id", PLACE_ID)
+      .put("name", RESTAURANT_NAME)
+      .put("vicinity", VICINITY)
+      .put("types", EMPTY_TYPES)
+      .put("geometry", geometry)
+      .put("rating", RATING)
+      .put("user_ratings_total", NUM_RATINGS)
+      .put("price_level", PRICE_LEVEL);
+
+    Restaurant expected = Restaurant.create(
+      PLACE_ID, RESTAURANT_NAME, VICINITY, COORDS, RATING, NUM_RATINGS,
+      PRICE_LEVEL, EMPTY_TYPES);
+    Restaurant actual = JsonToRestaurantParser.toRestaurant(body);
+    assertThat(actual).isEqualTo(expected);
+  }
+
+  @Test
+  public void extraneousTypes_onlyIncludeRelevantTypes() throws JSONException {
+    List<String> randomTypes = Arrays.asList(new String[]{"type a", "meal_takeaway", "type b", "meal_delivery", "restaurant"});
+    JSONObject geometry = new JSONObject().put("location", COORDS);
+    JSONObject body = new JSONObject()
+      .put("place_id", PLACE_ID)
+      .put("name", RESTAURANT_NAME)
+      .put("vicinity", VICINITY)
+      .put("types", randomTypes)
+      .put("geometry", geometry);
+
+    Restaurant actual = JsonToRestaurantParser.toRestaurant(body);
+    assertThat(actual.getPlaceTypes()).containsExactlyElementsIn(ALL_TYPES);
+  }
+
+  @Test
+  public void noOptionalFields_useDefaultValues() throws JSONException {
+    JSONObject geometry = new JSONObject().put("location", COORDS);
+    JSONObject body = new JSONObject()
+      .put("place_id", PLACE_ID)
+      .put("name", RESTAURANT_NAME)
+      .put("vicinity", VICINITY)
+      .put("types", EMPTY_TYPES)
+      .put("geometry", geometry);
+
+    Restaurant expected = Restaurant.create(
+      PLACE_ID, RESTAURANT_NAME, VICINITY, COORDS, /* avgRating= */ MISSING_FIELD, 
+      /* numRatings= */ MISSING_FIELD, /* priceNum= */ MISSING_FIELD, EMPTY_TYPES);
+    Restaurant actual = JsonToRestaurantParser.toRestaurant(body);
+    assertThat(actual).isEqualTo(expected);
+  }
+
+  @Test(expected = JSONException.class)
+  public void noFields() throws JSONException {
+    JSONObject body = new JSONObject();
+    Restaurant actual = JsonToRestaurantParser.toRestaurant(body);
+  }
+
+  @Test(expected = JSONException.class)
+  public void missingMandatoryField() throws JSONException {
+    JSONObject body = new JSONObject()
+      .put("place_id", PLACE_ID)
+      .put("name", RESTAURANT_NAME)
+      .put("vicinity", VICINITY)
+      .put("types", EMPTY_TYPES);
+    Restaurant actual = JsonToRestaurantParser.toRestaurant(body);
```

Comment:
nit: don't need to have actual declared as a local var -- just leaving it at JsonToRestaurantParser.toRestaurant(body);